From 4c18a0b6238f88e8f6deda6259e1d17968bcf755 Mon Sep 17 00:00:00 2001
From: Jan Beich <jbeich@FreeBSD.org>
Date: Tue, 27 Aug 2019 13:19:07 +0000
Subject: [PATCH] Unbreak build with fmtlib 6.0.0

In file included from ../src/factory.cpp:1:
In file included from ../include/factory.hpp:4:
../include/modules/clock.hpp:5:10: fatal error: 'fmt/time.h' file not found
 #include "fmt/time.h"
          ^~~~~~~~~~~~
In file included from ../src/bar.cpp:4:
In file included from ../include/factory.hpp:4:
In file included from ../include/modules/clock.hpp:3:
In file included from /usr/include/fmt/chrono.h:12:
/usr/include/fmt/locale.h:19:35: error: parameter type 'fmt::v5::internal::buffer' (aka 'basic_buffer<char>') is an abstract class
    const std::locale& loc, buffer<Char>& buf,
                                  ^
/usr/include/spdlog/fmt/bundled/core.h:238:16: note: unimplemented pure virtual method 'grow' in 'basic_buffer'
  virtual void grow(std::size_t capacity) = 0;
               ^
In file included from ../src/modules/sni/host.cpp:3:
/usr/include/fmt/ostream.h:22:9: error: expected member name or ';' after declaration specifiers
  buffer<Char>& buffer_;
  ~~~~~~^
/usr/include/fmt/ostream.h:25:19: error: expected ')'
  formatbuf(buffer<Char>& buf) : buffer_(buf) {}
                  ^
/usr/include/fmt/ostream.h:25:12: note: to match this '('
  formatbuf(buffer<Char>& buf) : buffer_(buf) {}
           ^
/usr/include/fmt/ostream.h:25:42: error: use of undeclared identifier 'buf'; did you mean 'prettify_handler::buf'?
  formatbuf(buffer<Char>& buf) : buffer_(buf) {}
                                         ^~~
                                         prettify_handler::buf
/usr/include/spdlog/fmt/bundled/format-inl.h:551:11: note: 'prettify_handler::buf' declared here
  buffer &buf;
          ^
---
 include/modules/clock.hpp | 2 +-
 src/bar.cpp               | 2 +-
 src/modules/sni/host.cpp  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/modules/clock.hpp b/include/modules/clock.hpp
index b812b70..4a0c1e5 100644
--- a/include/modules/clock.hpp
+++ b/include/modules/clock.hpp
@@ -1,8 +1,8 @@
 #pragma once
 
+#include <fmt/chrono.h>
 #include <fmt/format.h>
 #include "ALabel.hpp"
-#include "fmt/time.h"
 #include "util/sleeper_thread.hpp"
 
 namespace waybar::modules {
diff --git a/src/bar.cpp b/src/bar.cpp
index 8a45869..8e64349 100644
--- a/src/bar.cpp
+++ b/src/bar.cpp
@@ -1,7 +1,7 @@
 #include "bar.hpp"
-#include <spdlog/spdlog.h>
 #include "client.hpp"
 #include "factory.hpp"
+#include <spdlog/spdlog.h>
 
 waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
     : output(w_output),
diff --git a/src/modules/sni/host.cpp b/src/modules/sni/host.cpp
index 015f756..c815c65 100644
--- a/src/modules/sni/host.cpp
+++ b/src/modules/sni/host.cpp
@@ -1,6 +1,6 @@
 #include "modules/sni/host.hpp"
-#include <spdlog/spdlog.h>
 #include <fmt/ostream.h>
+#include <spdlog/spdlog.h>
 
 namespace waybar::modules::SNI {
 
